tests/buffer/buffer_get_count.cpp

```diff
@@ -10,17 +10,24 @@
 using namespace cl::sycl;
 
 int test_main(int argc, char *argv[]) {
-	std::vector<int> v(10);
-	size_t bufferSize = std::distance(v.begin(), v.end()); 
-	std::shared_ptr<int> ptr{new int[bufferSize]};
-	std::copy(v.begin(), v.end(), ptr.get());
-	buffer<int, 1> buf(ptr, range<1>(bufferSize));
+  std::vector<int> v(10);
+  size_t bufferSize = std::distance(v.begin(), v.end()); 
+  std::shared_ptr<int> ptr{new int[bufferSize]};
```

Comment:
I suspect the code to be wrong.
This is an unfortunate case where you hit http://www.acodersjourney.com/2016/05/top-10-dumb-mistakes-avoid-c-11-smart-pointers/
http://stackoverflow.com/questions/14957359/problems-with-shared-ptrt-wrapping-a-dynamic-array